database.go

```diff
@@ -1075,6 +1134,35 @@ func (db *database) addSeriesToIndex(measurementName string, s *Series) bool {
 	return idx.addSeries(s)
 }
 
+// dropSeries removes the series from the in memory references
+func (db *database) dropSeries(seriesByMeasurement map[string][]uint32) error {
+	for measurement, ids := range seriesByMeasurement {
+		for _, id := range ids {
+			// if the series is already gone, return
+			if db.series[id] == nil {
+				continue
+			}
+
+			delete(db.series, id)
+
+			// Remove series information from measurements
+			m := db.measurements[measurement]
+			if !m.dropSeries(id) {
+				return fmt.Errorf("failed to remove series id %d from measurment %q", id, m.Name)
```

Comment:
This shouldn't throw an error. If it was already removed, we're ok with that, just `continue` on the loop